.devcontainer/devcontainer.json

```diff
@@ -0,0 +1,49 @@
+// For format details, see https://aka.ms/devcontainer.json. For config options, see the README at:
+// https://github.com/microsoft/vscode-dev-containers/tree/v0.245.2/containers/cpp
+{
+    "name": "C++",
+    "build": {
+        "dockerfile": "Dockerfile",
+        // Update 'VARIANT' to pick an Debian / Ubuntu OS version: debian-11, debian-10, ubuntu-22.04, ubuntu-20.04, ubuntu-18.04
+        // Use Debian 11, Ubuntu 18.04 or Ubuntu 22.04 on local arm64/Apple Silicon
+        "args": {
+            "VARIANT": "ubuntu-22.04",
```

Comment:
`ARG`'s are user-providable arguments in Dockerfiles, you can give them during builds like:

```sh
docker build -t essearch/ess-elasticsearch:1.7.6 \
--build-arg number_of_shards=5 \
--build-arg number_of_replicas=2
```

When an `ARG` is declared, it can be given a default value

The values provided in the `devcontainer.json` -> `build.args` are passed to the Docker build API and take precedence over default values